### PR TITLE
Add overload to HealthChecks for configuring options

### LINKED ISSUE #8530: services.AddHealthChecks missing overload that takes delegate for configuring HealthChecksOptions
We need to add the overload to Health Checks to enable passing an options configuration delegate when adding Health Checks to services, e.g.:

``` csharp
services.AddHealthChecks(options =>
{
    options.AllowCachingResponses = true;
}).AddCheck<MyHealthCheck>()
```

@rynowak @glennc

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/dotnet/aspnetcore/issues/8530

### CODE REVIEW

File: src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs
Comment (pranavkm) [inline src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs:30]: Isn't it sufficient to call `services.Configure(configureOptions)`?

File: src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs
Comment (pranavkm) [inline src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs:42]: Why is this needed?

File: src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs
Comment (pranavkm) [reply src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs:30]: Could you add a unit test for this?

**Review by pranavkm — CHANGES_REQUESTED** [review]

File: src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs
Comment (Kahbazi) [reply src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs:42]: This overload allows to resolve a service from IoC container to configure the options. 
Several services now have both overloads. https://github.com/dotnet/aspnetcore/commit/5a375a7c226a3fa30720035dbc0796a664ae8a9b

Comment (HaoK) [reply src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs:42]: This doesn't seem worth adding, the underlying options builder api supports up to 5 services, and it's an uncommon enough scenario that it doesn't seem worth adopting as a general pattern to have all AddXyz methods add this overload.  I think we should encourage people who want to configure options with services to do it via optionsbuilder

Comment (Kahbazi) [reply src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs]: I removed this overload.

File: src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs
Comment (Kahbazi) [reply src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs]: I added a test.

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
@dougbu do we need to do anything to make the PublicAPI files to be regened for this project?

**Comment by dougbu** [comment]
@pranavkm yes, this PR was created before the project even had public API baselines. You'll need to open the project or a containing solution in VS, build and use the analyzer's fixer to add the new API to the baselines. Do not merge before updating because it'll merge cleanly and break every subsequent build.

There's more information in #24347

**Comment by pranavkm** [comment]
@dougbu I'm having no luck getting the analyzer to complain. Do I need to do enable something for it work?

**Comment by dougbu** [comment]
> @dougbu I'm having no luck getting the analyzer to complain. Do I need to do enable something for it work?

Nothing other than the steps I outlined should be necessary once you've built inside Visual Studio and looked at the errors and warnings for RS0016 (think that's the right one)

**Comment by weichch** [comment]
Frankly, I don't think this PR fixes https://github.com/dotnet/aspnetcore/issues/8530 and the issue doesn't look valid. https://github.com/dotnet/aspnetcore/issues/8530#issuecomment-703060414

**Comment by pranavkm** [comment]
@weichch you are right. @Kahbazi I'm going to close this since a fix for this is much more different than what's being done here.

**Pull request closed without merging**
